tests-functional/tests/test_wakuext_message_reactions.py

```diff
@@ -106,3 +107,71 @@ def test_one_to_one_message_reactions(self, backend_new_profile, waku_light_clie
                     item["emoji"] == "🙂",
                 )
             )
+
+    @pytest.mark.parametrize("waku_light_client", [False], indirect=True, ids=["wakuV2LightClient_False"])
+    def test_limit_of_20_reactions(self, backend_new_profile, waku_light_client):
+        """Test that you cannot send more than 20 message reactions on a single message"""
+        # Initialize two backends (sender and receiver) for this test
+        self.sender = backend_new_profile("sender", waku_light_client=waku_light_client)
+        self.receiver = backend_new_profile("receiver", waku_light_client=waku_light_client)
+
+        self.make_contacts(self.sender, self.receiver)
+        response = self.sender.wakuext_service.send_one_to_one_message(self.receiver.public_key, "test_message")
+        message = self.get_message_by_content_type(response, content_type=MessageContentType.TEXT_PLAIN.value)[0]
+        message_id, sender_chat_id = message["id"], message["chatId"]
+        receiver_chat_id = self.receiver.wakuext_service.rpc_request(method="chats")[0]["id"]
+
+        # Send 20 emojis
+        emojis = ["😀", "🤓", "😄", "😁", "😆", "😅", "🤣", "😂", "🥹", "🙂", "🙃", "😉", "😊", "😇", "🥰", "😍", "🤩", "😘", "😗", "☺️"]
+        last_emoji_id = None
+        for emoji in emojis:
+            response = self.sender.wakuext_service.rpc_request(method="sendEmojiReactionV2", params=[sender_chat_id, message_id, emoji])
+            assert response["emojiReactions"][0]["emoji"] == emoji
+            last_emoji_id = response["emojiReactions"][0]["id"]
+
+        # The 21st emoji sent should fail
+        try:
+            _ = self.sender.wakuext_service.rpc_request(method="sendEmojiReactionV2", params=[sender_chat_id, message_id, "😋"])
+            assert False, "Expected an error when sending the 21st emoji reaction, but the request succeeded."
+        except ApiResponseError as err:
+            assert "too many emoji reactions for message" in str(err)
+
+        # Test not processing 21st emoji
+        # We don't fetch the messages from the receiver, we'll just try to send more reactions and see that they fail for the og sender
+        response = self.receiver.wakuext_service.rpc_request(method="sendEmojiReactionV2", params=[receiver_chat_id, message_id, "🙏"])
+        emoji_1_id = response["emojiReactions"][0]["id"]
+        assert response["emojiReactions"][0]["emoji"] == "🙏"  # This one works because the receiver doesn't know about the other reactions
+
+        try:
+            self.sender.find_signal_containing_pattern(
+                SignalType.MESSAGES_NEW.value,
+                event_pattern=emoji_1_id,
+                timeout=10,
+            )
+            assert False, "Expected not to receive a signal for the 21st reaction, but did."
```

Comment:
Waiting for absence of smth is an anti-pattern 🙁 
https://www.notion.so/Prevent-flaky-tests-2045334180cb4eabbe2151f7005e16ce

I guess there's no way to implement such test in a reliable way. We would need to pause the `RetrieveMessagesLoop` somehow:
https://github.com/status-im/status-go/blob/ce85b74e6cb63bcd50772dc3f721b31498cb4b49/protocol/messenger.go#L2691-L2698

We could implement such an endpoint though `StopRetrieveMessagesLoop`, and then `StartRetrieveMessagesLoop`. This sounds reasonable to me, especially for testing purposes. But I can even imagine use cases for production.

@osmaczko wdyt?

Reply:
I'll remove that part of the test for now. Functional tests are not the best place to check for that anyway, since there is no signal that we skipped a message. I'll improve the integration test instead